### PR TITLE
Bivariate branch

### PR DESCRIPTION
**Pull request by yuqizhang99**
## Functionality

Create bivariate chart components for Q+C Q+T

## Issues addressed

#95

### CODE REVIEW

**Comment by yuqizhang99** [comment]
The current branch is up to date except for the Q+T bivariate chart

File: .github/pull_request_template.md
Comment (willeppy) [inline .github/pull_request_template.md:1]: Remove this file from PR

File: .github/screenshots/profiler_sc.png
Comment (willeppy) [inline .github/screenshots/profiler_sc.png:1]: Remove from PR

File: examples/SF Housing Demo.ipynb
Comment (willeppy) [inline examples/SF Housing Demo.ipynb:1]: Remove from PR

**Review by willeppy — CHANGES_REQUESTED** [review]
# Design
Plus button for add chart
- [x] Move to bottom right hand side
- [ ] Maybe make wider (not super important)
- [x] Make sure that the hover only happens on the actual button, not the whole div row
- [x] Should always be below newly added charts!!!

Column drop downs
- [ ] add left padding so not so tight
- [ ] drop down should disappear when you click outside
- [ ] Is it possible to make the seach stay inline rather than at the top? Like [this example](https://www.w3schools.com/howto/howto_js_filter_dropdown.asp)
- [ ] OR just use a normal dropdown and we can worry about searchability later (my vote!)
- [ ] Can you add icon for the column data type in the dropdown?

Added chart tab
- [ ] Make the same width, height, and color as normal column button
- [ ] Icon is cluttered
- [ ] Change aggregation selection to be normal dropdown (Look at the one for Sort by at the top)
- [x] What does "---" mean in axis?

Bar chart
- [ ] Make y axis wider so can more easily read values
- [ ] Is it possible to show a tooltip on hover for each bar with the x and y axis values?
- [ ] Sort y axis by default as descending
- [ ] add sorting for y axis

Time chart
- [ ] Changing time agg to second, minute, hour overplots on x axis like crazy
- [ ] Make dropdowns normal selects here as well (like sort by)

For charts tab
- [ ] Change delete to trash can icon

# Authoring Flow
- [ ] When I hit add to chart, the plus button should disappear until I have hit done or cancel; OR change the plus button to say "cancel" once it is clicked (I vote for disappear though)
- [ ] Do not allow adding same chart twice: right now, I can add same chart twice (county + year, year + county)
- [ ] Add some feedback text for what is being filtered in selectors. E.g. if I first select a Q column, it should say "Only showing nominal columns" in second dropdown. This might not be necessary if include data types in dropdown though
- [ ] done should be disabled until I have selected both dropdowns
- [ ] Disable second dropdown until first is selected

# Testing
- [ ] Be sure to test edge cases like if dataframe is completley empty (freezes up right now)
- [ ] Test that it works if one (or both) of the columns is all null 
- [ ] BUG: If first selected column is Q then cant select second dropdown

**Comment by yuqizhang99** [comment]
Some bugs required to fix:
- [ ] After clicking "Add" button, there should be an option for users to cancel this interaction
- [ ] The dropdown menu should have limited width

**Comment by yuqizhang99** [comment]
Sorry I force-pushed the code because I squashed the commits to the commit behind the remote.

**Comment by willeppy** [comment]
Final TODOs
- [ ] Change done button to be done or cancel
- [ ] add button for delete
- [ ] make edit button look nicer
- [ ] clean up code

**Review by willeppy — APPROVED** [review]

**Pull request merged**